### PR TITLE
Avoid using Authorization headers in GET requests to avoid CORS preflight

### PR DESCRIPTION
**Pull request by maxfierke**
According to the CORS spec, the `Authorization` header is not a [simple header](https://www.w3.org/TR/cors/#simple-header), so requests including it are subject to preflight. It's a weird, seemingly arbitrary distinction, but we can get around it by putting our OAuth access tokens in the `access_token` query string parameter and then the browser won't do a preflight request. Requires that the backend supports doing this.

`¯\_(ツ)_/¯`

### CODE REVIEW

File: examples/output/adapters/v1/calendar-event.js [hunk 1]
```diff
@@ -11,11 +11,17 @@ export default DS.RESTAdapter.extend({
     return Ember.String.pluralize(underscorized);
   },
 
-  headers: function() {
-    return {
-      'AUTHORIZATION': 'Bearer ' + this.get('token')
-    };
-  }.property('token'),
+  ajaxOptions: function(url, type, options) {
```
Comment (bfitch) [inline examples/output/adapters/v1/calendar-event.js]: You could do some fancy, schmasy ES2015 destructuring here:
```js
    ajaxOptions: function(url, type, {data={},headers={},...opts} = {}) {
     if (type === "GET") {
       data["access_token"] = this.get('token');
     } else {
       headers["Authorization"] = `Bearer ${this.get('token')}`;
     }
     Object.assign(opts,{headers,data});
     return this._super(url, type, opts);
  }
```

Comment (bfitch) [inline examples/output/adapters/v1/calendar-event.js]: oh...these are auto-generated. nevermind.

**Comment by bfitch** [comment]
:octocat:

File: examples/output/adapters/v1/calendar-event.js (hunk 1 above)
Comment (patricksrobertson) [inline examples/output/adapters/v1/calendar-event.js]: Can we assume that all apps including the code have ES2015 as transpilation target?

**Comment by patricksrobertson** [comment]
Fantastic sleuthing @maxfierke! Will merge and release new version of braise today!

**Comment by patricksrobertson** [comment]
Also, the authentication module we use for our services by default checks both the query params and the auth header. If we have something that isn't doing it -- we should standardize it on the common module.

**Pull request merged**